h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/rabit/RabitTrackerH2O.java

```diff
@@ -113,9 +131,10 @@ public void run() {
             List<RabitWorker> pending = new ArrayList<>();
             Queue<Integer> todoNodes = new ArrayDeque<>(tracker.workers);
             while (!interrupted() && shutdown.size() != tracker.workers) {
-                try {
-                    SocketChannel channel = tracker.sock.accept();
-                    RabitWorker worker = new RabitWorker(channel);
+                try{
+                    final SocketChannel channel = tracker.sock.accept();
+                    socketChannels.add(channel);
```

Comment:
This modifies the state of the tracker. It doesn't seem like the list is every reset to its original state anywhere (eg. in the stop()) method.

Reply:
I believed these are Garbage collected, even though nothing prevents us from empyting & nullifying these anyway. 
The thread is not reused.

In tracker's stop method, the reference is removed from the variable, thus the whole thread is garbage collected after interrupted.

```java
    @Override
    public void stop() {
        if(null != this.trackerThread) {
            this.trackerThread.interrupt();
            this.trackerThread = null; <--- Here. Never reused.

            try {
                this.sock.close();
            } catch (IOException e) {
                throw new RuntimeException("Failed to close Rabit tracker socket.", e);
            } finally {
                this.port = 9091;
            }
        }
    }
```